fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -631,3 +562,89 @@ def loop_over_hessians(controller, options, minimizer_name, grabbed_output):
             break
 
     return new_result, new_chi_sq, new_minimizer_list
+
+
+def perform_fit(controller, num_runs, grabbed_output):
```

Comment:
It would be nice if this returned a `FittingResult`, or the `result_args` which get turned into Fitting results at the loop_over_minimizers level

Reply:
Could we change this in a future PR? Moving the `results_args` into this function will make the function more complicated with needing to pass in the `minimizer`, `minimizer_name_hes`, `problem` and `cost_func`, and I want to try keep this function smallish

Reply:
Sure, `minimizer`, `problem` and `cost_func` all come from controller, so we're already passing them but it would make it bigger and possibly push it into pylints compliants again!